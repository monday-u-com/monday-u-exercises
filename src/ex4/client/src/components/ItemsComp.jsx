import "../App.css";
import ItemComp from "./ItemComp";

function ItemsComp({ deleteItem, updateItem, allItems }) {
	const dismantleAllItems = () => {
		return (
			<div>
				{allItems.map((item) => {
					return (
						<ItemComp
							key={item.id}
							item={item}
							deleteItem={async () => await deleteItem(item.id)}
							updateItem={updateItem}
						/>
					);
				})}
			</div>
		);
	};

	return dismantleAllItems();
}

export default ItemsComp;
