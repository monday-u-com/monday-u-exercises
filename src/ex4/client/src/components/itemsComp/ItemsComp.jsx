import { isInteger } from "lodash";
import { useEffect, useState } from "react";
import "../../App.css";
import ItemComp from "../itemComp/ItemComp";

function ItemsComp({ items, deleteItemAction, updateItemAction, searchText }) {
	const [filteredItems, setFilteredItems] = useState(items);

	useEffect(() => {
		if (searchText && !isInteger(parseInt(searchText))) {
			const filteredItems = items.filter((item) =>
				item.itemName.includes(searchText)
			);
			setFilteredItems(filteredItems);
		} else if (isInteger(parseInt(searchText))) {
			const filteredItems = items.filter(
				(item) => item.pokedexId === parseInt(searchText)
			);
			setFilteredItems(filteredItems);
		} else {
			setFilteredItems(items);
		}
	}, [items, searchText]);

	return (
		<div>
			{filteredItems.map((item) => {
				return (
					<ItemComp
						key={item.id}
						item={item}
						deleteItemAction={deleteItemAction}
						updateItemAction={updateItemAction}
					/>
				);
			})}
		</div>
	);
}

export default ItemsComp;
