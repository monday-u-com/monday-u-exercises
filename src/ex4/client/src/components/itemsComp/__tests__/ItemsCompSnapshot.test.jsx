import { create, act } from "react-test-renderer";
import ItemsComp from "../ItemsComp";

const items = [
	{
		id: 56,
		itemName: "Take dog out for a walk",
		status: false,
		pokedexId: 25,
	},
	{
		id: 32,
		itemName: "Do the dishes",
		status: true,
		pokedexId: 1,
	},
];

test("renders currectly items Array item must include id,itemName,status,pokedexId", () => {
	let root;
	act(() => {
		root = create(<ItemsComp items={items} searchText={""} />);
	});
	expect(root.toJSON()).toMatchSnapshot();

	act(() => {
		root.update(<ItemsComp items={items} searchText={items[1].itemName} />);
	});
	expect(root.toJSON()).toMatchSnapshot();

	act(() => {
		root.update(<ItemsComp items={items} searchText={items[0].pokedexId} />);
	});
	expect(root.toJSON()).toMatchSnapshot();
});
