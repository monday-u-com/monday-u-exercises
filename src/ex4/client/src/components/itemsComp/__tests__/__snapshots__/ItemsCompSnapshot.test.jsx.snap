// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders currectly items Array item must include id,itemName,status,pokedexId 1`] = `
<div>
  <div
    className="item-container"
  >
    <label
      className="monday-style-checkbox checkbox"
      onClickCapture={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={false}
        className="monday-style-checkbox__input"
        disabled={false}
        name=""
        onChange={[Function]}
        type="checkbox"
        value=""
      />
      <div
        className="monday-style-checkbox__checkbox monday-style-checkbox__prevent-animation"
      >
        <svg
          aria-hidden={true}
          className="icon_component monday-style-checkbox__icon icon_component--no-focus-style"
          fill="currentColor"
          height="16"
          onClick={[Function]}
          viewBox="0 0 20 20"
          width="16"
        >
          <path
            clipRule="evenodd"
            d="M8.53033 14.2478L8 13.7175L7.46967 14.2478C7.76256 14.5407 8.23744 14.5407 8.53033 14.2478ZM8 12.6569L4.53033 9.18718C4.23744 8.89429 3.76256 8.89429 3.46967 9.18718C3.17678 9.48008 3.17678 9.95495 3.46967 10.2478L7.46967 14.2478L8 13.7175L8.53033 14.2478L16.2478 6.53033C16.5407 6.23743 16.5407 5.76256 16.2478 5.46967C15.955 5.17677 15.4801 5.17677 15.1872 5.46967L8 12.6569Z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <span
        className="monday-style-checkbox__label"
      />
    </label>
    <div
      className="clickable task-element"
      onClick={[Function]}
    >
      Take dog out for a walk
      <div
        className="clickable delete-button"
        onClick={[Function]}
      >
        <img
          className="tash-icon"
          src="trash.png"
        />
      </div>
    </div>
  </div>
  <div
    className="item-container"
  >
    <label
      className="monday-style-checkbox checkbox"
      onClickCapture={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={true}
        className="monday-style-checkbox__input"
        disabled={false}
        name=""
        onChange={[Function]}
        type="checkbox"
        value=""
      />
      <div
        className="monday-style-checkbox__checkbox monday-style-checkbox__prevent-animation"
      >
        <svg
          aria-hidden={true}
          className="icon_component monday-style-checkbox__icon icon_component--no-focus-style"
          fill="currentColor"
          height="16"
          onClick={[Function]}
          viewBox="0 0 20 20"
          width="16"
        >
          <path
            clipRule="evenodd"
            d="M8.53033 14.2478L8 13.7175L7.46967 14.2478C7.76256 14.5407 8.23744 14.5407 8.53033 14.2478ZM8 12.6569L4.53033 9.18718C4.23744 8.89429 3.76256 8.89429 3.46967 9.18718C3.17678 9.48008 3.17678 9.95495 3.46967 10.2478L7.46967 14.2478L8 13.7175L8.53033 14.2478L16.2478 6.53033C16.5407 6.23743 16.5407 5.76256 16.2478 5.46967C15.955 5.17677 15.4801 5.17677 15.1872 5.46967L8 12.6569Z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <span
        className="monday-style-checkbox__label"
      />
    </label>
    <div
      className="clickable task-element"
      onClick={[Function]}
    >
      Do the dishes
      <div
        className="clickable delete-button"
        onClick={[Function]}
      >
        <img
          className="tash-icon"
          src="trash.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders currectly items Array item must include id,itemName,status,pokedexId 2`] = `
<div>
  <div
    className="item-container"
  >
    <label
      className="monday-style-checkbox checkbox"
      onClickCapture={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={true}
        className="monday-style-checkbox__input"
        disabled={false}
        name=""
        onChange={[Function]}
        type="checkbox"
        value=""
      />
      <div
        className="monday-style-checkbox__checkbox monday-style-checkbox__prevent-animation"
      >
        <svg
          aria-hidden={true}
          className="icon_component monday-style-checkbox__icon icon_component--no-focus-style"
          fill="currentColor"
          height="16"
          onClick={[Function]}
          viewBox="0 0 20 20"
          width="16"
        >
          <path
            clipRule="evenodd"
            d="M8.53033 14.2478L8 13.7175L7.46967 14.2478C7.76256 14.5407 8.23744 14.5407 8.53033 14.2478ZM8 12.6569L4.53033 9.18718C4.23744 8.89429 3.76256 8.89429 3.46967 9.18718C3.17678 9.48008 3.17678 9.95495 3.46967 10.2478L7.46967 14.2478L8 13.7175L8.53033 14.2478L16.2478 6.53033C16.5407 6.23743 16.5407 5.76256 16.2478 5.46967C15.955 5.17677 15.4801 5.17677 15.1872 5.46967L8 12.6569Z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <span
        className="monday-style-checkbox__label"
      />
    </label>
    <div
      className="clickable task-element"
      onClick={[Function]}
    >
      Do the dishes
      <div
        className="clickable delete-button"
        onClick={[Function]}
      >
        <img
          className="tash-icon"
          src="trash.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders currectly items Array item must include id,itemName,status,pokedexId 3`] = `
<div>
  <div
    className="item-container"
  >
    <label
      className="monday-style-checkbox checkbox"
      onClickCapture={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={false}
        className="monday-style-checkbox__input"
        disabled={false}
        name=""
        onChange={[Function]}
        type="checkbox"
        value=""
      />
      <div
        className="monday-style-checkbox__checkbox monday-style-checkbox__prevent-animation"
      >
        <svg
          aria-hidden={true}
          className="icon_component monday-style-checkbox__icon icon_component--no-focus-style"
          fill="currentColor"
          height="16"
          onClick={[Function]}
          viewBox="0 0 20 20"
          width="16"
        >
          <path
            clipRule="evenodd"
            d="M8.53033 14.2478L8 13.7175L7.46967 14.2478C7.76256 14.5407 8.23744 14.5407 8.53033 14.2478ZM8 12.6569L4.53033 9.18718C4.23744 8.89429 3.76256 8.89429 3.46967 9.18718C3.17678 9.48008 3.17678 9.95495 3.46967 10.2478L7.46967 14.2478L8 13.7175L8.53033 14.2478L16.2478 6.53033C16.5407 6.23743 16.5407 5.76256 16.2478 5.46967C15.955 5.17677 15.4801 5.17677 15.1872 5.46967L8 12.6569Z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <span
        className="monday-style-checkbox__label"
      />
    </label>
    <div
      className="clickable task-element"
      onClick={[Function]}
    >
      Take dog out for a walk
      <div
        className="clickable delete-button"
        onClick={[Function]}
      >
        <img
          className="tash-icon"
          src="trash.png"
        />
      </div>
    </div>
  </div>
</div>
`;
